ECE444-2023Fall/Blue-Surf
# Fixing the update event for all missing fields (date, club)

Closes #106 
Previously, editing dates and clubs were not reflected in the database. This has now been fixed with this PR. 
To test this change, create a post, publish it, then edit the date. When you go back to the landing page and reopen the post, the newly updated date should be visible. The other validations (start time < end time) should also remain intact. 

Additionally, there were tests added at the datalayer level (in `test_datalayer_event.py`) and the API level (in `test_app.py`) to cover all cases. 

I also manually did regression testing to make sure that all previous functionality remains intact.

To test this:

1. Delete the `bluesurf-postgresql` file, then uncomment the code in `app.py` to make sure the database is populated and created initially (if bluesurf-postgresql doesn't already exist).
2. Start the backend using `npm run start-backend`
3. Start the frontend in a separate terminal using `npm start`
4. Create a post.
5. Edit the start time of the post. Go back to the landing page and validate that the previewed time is the new time.
6. In the terminal, run `pytest` and ensure that all tests pass. If some fail, try running it again, sometimes there are issues with the database setup.
